ingestor/src/main/scala/weco/concepts/ingestor/json/JsonOps.scala

```diff
@@ -0,0 +1,39 @@
+package weco.concepts.ingestor.json
+
+import scala.util.Try
+
+trait JsonOption[T] {
+  def apply(node: ujson.Value): Option[T]
+}
+
+object JsonOps {
```

Comment:
I thought it would be interesting to try a different approach to JSON than what we've done previously with Circe. I'm using the `ujson` subset of https://github.com/com-lihaoyi/upickle to traverse the JSON based on specific fields - as one might in JavaScript - rather than by modelling the whole thing as case classes which get slurped up (slowly) by shapeless. This file contains a typeclass and some helpers for making that traversal easier, as `ujson`'s API doesn't handle optional fields very naturally.

Reply:
I wonder if this eventually belongs in a common library like the models.

Reply:
Yes, agreed